js/binance.js

```diff
@@ -1777,16 +1855,15 @@ module.exports = class binance extends Exchange {
             const base = this.safeCurrencyCode (baseId);
             const quote = this.safeCurrencyCode (quoteId);
             const settle = this.safeCurrencyCode (settleId);
-            const contract = future || delivery;
             const contractType = this.safeString (market, 'contractType');
-            const idSymbol = contract && (contractType !== 'PERPETUAL');
-            let symbol = undefined;
-            let expiry = undefined;
-            if (idSymbol) {
-                symbol = id;
-                expiry = this.safeInteger (market, 'deliveryDate');
+            const contract = ('contractType' in market);
+            const spot = !contract;
+            let expiry = this.safeInteger (market, 'deliveryDate');
+            if ((contractType === 'PERPETUAL') || (expiry === 4133404800000)) { // some swap markets do not have contract type, eg: BTCST
```

Comment:
this is not a bug but I still believe it is better to use other means to define it instead of comparing to that integer number 4133404800000.  That seems to be their 'magic-constant' , but is it not an officially documented way by binance, and I doubt they may change that integer at some day, causing massive trade-issues or even losses for ccxt users on that particular day. 
So, i think it's much better to use other ways to make it more future-proof , i.e.:
- `contractStatus, contractSize, equalQtyPrecision, maxMoveOrderLimit` <---  these are the keys that are only present in `dapi` symbols, not in `fapi`. also `dapi` symbols have `_PERP` in the end of symbol
- `status, settlePlan` <---  these are the keys that are only present in `fapi` symbols, not in `dapi`
maybe you can give it a second thought.